### DISCUSSION TITLE
Multi-paragraph footnotes. Footnotes separated from body text.

### DISCUSSION THREAD

**Discussion by johnmapeson** [typst/typst#4910] [Q&A] [answered]
The [`footnote`](https://typst.app/docs/reference/model/footnote/) page doesn't mention multi-paragraph footnotes. Here is what I use currently:

```
Lorem ipsum#footnote[
Footnote paragraph 1.

Footnote paragraph 2.
]
dolor sit amet.
```

This works for me, but since the `footnote` page doesn't mention multi-paragraph footnotes, is it correct to say that handling multi-paragraph footnotes is not really established yet, and somewhat work-in-process instead?

Also, is there a way to separate footnotes from body text? For example, like in Pandoc's Markdown:

```
Lorem ipsum[^fn1] dolor sit amet.

[^fn1]:
    Footnote paragraph 1.

    Footnote paragraph 2.
```

I know Typst has `#let` and footnote labels, but I didn't successed to use them for this.

**Comment by MDLC01** [comment]
Multi-paragraph footnotes are supported. The problem is that the default styling for footnote does not look very good when the body contains multiple paragraphs. You can use a show rule on [`footnote.entry`](https://typst.app/docs/reference/model/footnote/#definitions-entry) to change the appearance of the note. For example, you can try something like the following (of course, feel free to modify this to suit your needs):

```typ
#show footnote.entry: it => {
  let (number, ) = counter(footnote).at(it.note.location())
  grid(
    columns: 2,
    gutter: 0.5em,
    numbering(it.note.numbering, number),
    it.note.body,
  )
}
```

**Comment by johnmapeson** [comment]
Hello, Malo. Thanks for the snippet.

- "Multi-paragraph footnotes are supported." - Am I use them correctly, or there is a better way?

- And maybe you know, is there a way to separate footnotes from body text, like in my Pandoc's Markdown example?

**Reply by MDLC01** [reply, marked as answer]
> "Multi-paragraph footnotes are supported." - Am I use them correctly, or there is a better way?

You are using them properly.

> is there a way to separate footnotes from body text, like in my Pandoc's Markdown example?

You can first define the footnote element and assign it to a variable, and then use it.

```typ
#let fn = footnote[My footnote.]
#lorem(10)#fn
```

I don't think you can do it the other way around, like in your example.
